mountpoint-s3-client/src/object_client.rs

```diff
@@ -245,9 +245,19 @@ pub enum GetObjectAttributesError {
 
 /// Parameters to a [ObjectClient::put_object] request
 /// TODO: Populate this struct with parameters from the S3 API, e.g., storage class, encryption.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Clone)]
 #[non_exhaustive]
-pub struct PutObjectParams {}
+pub struct PutObjectParams {
+    pub trailing_checksums: bool,
```

Comment:
Needs a doc comment (specifically that it's crc32c).

Reply:
Actually, maybe this should be `pub checksum_config: Option<ChecksumConfig>` and we re-export `mountpoint_s3_crt::ChecksumConfig` from this crate, so that we don't end up duplicating stuff if we add more options here.

Reply:
I thought about that, but it's unclear to me whether we want to expose ChecksumConfig in the long run (vs something tailored to specific requests: PutObject, MPU, GetObject).
I'd rather keep it as a simple bool for now and think about a more complete API when/if we'll introduce more options.

Reply:
I'd probably second having an enum/struct over boolean. Something like `Option<ChecksumAlgorithm>`, even if `ChecksumAlgorithm` will only be `CRC32C` today?

I'm thinking the API we're putting here matches up to this one in Java SDK: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/PutObjectRequest.Builder.html#checksumAlgorithm(software.amazon.awssdk.services.s3.model.ChecksumAlgorithm)

We don't _have_ to re-export the CRT config struct, we could just use our own in `mountpoint-s3-client` if it makes more sense.

Reply:
@dannycjones, we could start adding a `ChecksumAlgorithm` enum, but what about the location (trailer vs header)? That's part of `ChecksumConfig` but (I believe) only trailer is supported on multi-part uploads (which is what `put_object` uses). Do we want to limit the API to what a specific request allows, or just error on misconfiguration? Related: will we have a separate `put_object` for non-mpu? Or maybe that would also be configurable?

I don't think we **need** to answer any of these questions now, so my preference is for the simplest possible option, knowing we will replace it later.

Reply:
I probably prefer to keep it simple for now, so I'm ok with using boolean. Maybe we open a new issue to review this configuration before we publish a new version of mountpoint-s3-client? I think that's the point where we say there will be no breaking change after this and you can safely use this new config.